tests/test_models/test_losses/test_ce_loss.py

```diff
@@ -15,19 +18,56 @@ def test_ce_loss():
             loss_weight=1.0)
         build_loss(loss_cfg)
 
-    # test loss with class weights
+    # test loss with simple case for ce/bce
+    fake_pred = torch.Tensor([[100, -100]])
+    fake_label = torch.Tensor([1]).long()
     loss_cls_cfg = dict(
         type='CrossEntropyLoss',
-        use_sigmoid=False,
-        class_weight=[0.8, 0.2],
+        use_sigmoid=use_sigmoid,
         loss_weight=1.0,
+        avg_non_ignore=avg_non_ignore,
         loss_name='loss_ce')
     loss_cls = build_loss(loss_cls_cfg)
-    fake_pred = torch.Tensor([[100, -100]])
-    fake_label = torch.Tensor([1]).long()
-    assert torch.allclose(loss_cls(fake_pred, fake_label), torch.tensor(40.))
+    if use_sigmoid:
+        assert torch.allclose(
+            loss_cls(fake_pred, fake_label), torch.tensor(100.))
+    else:
+        assert torch.allclose(
+            loss_cls(fake_pred, fake_label), torch.tensor(200.))
+
+    # test loss with complicated case for ce/bce
+    # when avg_non_ignore is False, `avg_factor` would not be calculated
+    fake_pred = torch.full(size=(2, 21, 8, 8), fill_value=0.5)
+    fake_label = torch.ones(2, 8, 8).long()
+    fake_label[:, 0, 0] = 255
+    if use_sigmoid:
+        if avg_non_ignore:
+            assert torch.allclose(
+                loss_cls(fake_pred, fake_label, ignore_index=255),
+                torch.tensor(0.9503),
+                atol=1e-4)
+        else:
+            loss_cls = build_loss(loss_cls_cfg)
+            assert torch.allclose(
+                loss_cls(fake_pred, fake_label, ignore_index=255),
+                torch.tensor(0.9354),
+                atol=1e-4)
+    else:
+        if avg_non_ignore:
+            assert torch.allclose(
+                loss_cls(fake_pred, fake_label, ignore_index=255),
+                torch.tensor(3.0445),
+                atol=1e-4)
+        else:
+            loss_cls = build_loss(loss_cls_cfg)
+            assert torch.allclose(
+                loss_cls(fake_pred, fake_label, ignore_index=255),
+                torch.tensor(2.9970),
+                atol=1e-4)
```

Comment:
```suggestion
    if use_sigmoid:
        if avg_non_ignore:
            torch_loss = torch.nn.functional.binary_cross_entropy_with_logits(
                fake_pred, fake_label, reduction='mean', pos_weight=class_weight)
        else:
            torch_loss = torch.nn.functional.binary_cross_entropy_with_logits(
                fake_pred, fake_label, reduction='sum', pos_weight=class_weight)/xxxx
    else:
        if avg_non_ignore:
            torch_loss = torch.nn.functional.cross_entropy(
                fake_pred, fake_label, reduction='mean')
        else:
            torch_loss = torch.nn.functional.cross_entropy(
                fake_pred, fake_label, reduction='mean')
```
assert torch.allclose(loss, torch_loss)